### PR TITLE
TE-1379 Implement styles to handle dropdown ellipsis

### PR DESCRIPTION
**Pull request by red-runner**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-1379)

### What **one** thing does this PR do?
Implements the missing styles for the dropdowns. It's meant to fix `text ellipsis` and `width` issues
### Any other notes

basic dropdown no icon
dropdown_above_input
dropdown_disabled
dropdown_error
dropdown_valid_with_icon
dropdown_valid
dropdown_variation_icon
dropdown_variation_image
dropdown_variations_intendedoptions
dropdown_search
dropdpwn_rates

### CODE REVIEW

File: src/styles/semantic/themes/livingstone/globals/site.variables [hunk 1]
```diff
@@ -441,6 +441,8 @@
 @searchBarFixedZIndex: 999;
 @searchBarFixedPadding: @20px 0;
 @searchBarFixedBoxShadow: 0 -(@4px) @9px -(@6px) rgba(0, 0, 0, 0.29);
+@searchDropdownIconTotalWidth: 75px;
+@searchDropdownMinWidth: 90px;
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/globals/site.variables]: Can you keep these variables consistent? Such as:
```
@searchBarDropdownIconWidth: 75px;
@searchBarDropdownMinWidth: 90px;
```

File: src/styles/semantic/themes/livingstone/globals/site.variables [hunk 2]
```diff
@@ -441,6 +441,8 @@
 @searchBarFixedZIndex: 999;
 @searchBarFixedPadding: @20px 0;
 @searchBarFixedBoxShadow: 0 -(@4px) @9px -(@6px) rgba(0, 0, 0, 0.29);
+@searchBarDropdownIconWidth: 75px;
+@searchDropdownMinWidth: 90px;
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/globals/site.variables]: Can you also update the other variable? `@searchBarDropdownMinWidth `

File: src/styles/semantic/themes/livingstone/globals/site.variables (hunk 1 above)
Comment (stephenkr) [reply src/styles/semantic/themes/livingstone/globals/site.variables]: Can you also update the other variable? @searchBarDropdownMinWidth

File: src/styles/semantic/themes/livingstone/globals/site.overrides [hunk 3]
```diff
@@ -436,6 +436,42 @@ blockquote.ui.quote .row {
...
   }
+
+  .ui.dropdown {
+
+    > .text {
+      width: 100%;
+    }
+
+    + label {
+      width: ~"calc(100% - @{searchDropdownIconTotalWidth})";
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/globals/site.overrides]: Also this variable needs to be updated.

File: src/styles/semantic/themes/livingstone/globals/site.overrides [hunk 4]
```diff
@@ -436,6 +436,42 @@ blockquote.ui.quote .row {
...
+    + label,
+    &.selection .menu > .item {
+      white-space: nowrap;
+      text-overflow: ellipsis;
+      overflow: hidden;
+    }
+  }
+
+  .dropdown-container {
+    min-width: searchBarDropdownMinWidth;
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/globals/site.overrides]: You're missing the `@` prefix to make this a variable.

**Comment by marcdavi-es** [comment]
@red-runner Is it possible to solve this for all dropdowns, rather than only those nested in search bar?

**Comment by red-runner** [comment]
> @red-runner Is it possible to solve this for all dropdowns, rather than only those nested in search bar?

@marcdavi-es It would be good to keep improving this for the rest of the dropdowns. However, the search dropdowns have 2 icons instead of one making them a bit different. I'd personally merge this for the `search` and then go to `dropdown.overrides` / `dropdown.variables` and add ellipsis for all cases on a separate ticket.  

The styles added on this PR were thought for a structure of 4 inline dropdowns, which makes its variables specific for this case; **I wouldn't modify them**. Adding ellipsis for the rest of them should easy.

File: src/styles/semantic/themes/livingstone/modules/dropdown.overrides [hunk 5]
```diff
@@ -183,17 +208,30 @@
...
+
+      .dirty& {
+        width: 100%;
+      }
+    }
   }
 
   /*  With Image */
   &.has-images {
+    min-width: 118px;
```
Comment (Jonnygwi) [inline src/styles/semantic/themes/livingstone/modules/dropdown.overrides]: Where is this value coming from?

File: src/styles/semantic/themes/livingstone/modules/dropdown.variables [hunk 6]
```diff
@@ -165,17 +170,22 @@
 
 @errorMessageFontWeight: 400;
 @errorMessagePadding: @6px @25px;
-@errorMessageRightPosition: -(@1px);
+@errorMessageRightPosition: -(@2px);
 @errorMessageTopPosition: -(@30px);
 
 /* Valid */
+@dropdownValidMinWidth: 126px;
```
Comment (Jonnygwi) [inline src/styles/semantic/themes/livingstone/modules/dropdown.variables]: Same with this one: https://github.com/lodgify/lodgify-ui/pull/425/files#r231571067

File: src/styles/semantic/themes/livingstone/modules/dropdown.overrides (hunk 5 above)
Comment (red-runner) [reply src/styles/semantic/themes/livingstone/modules/dropdown.overrides]: @Jonnygwi This is the breackpoint where the text remains as just `...` and not just one character like `L...`

File: src/styles/semantic/themes/livingstone/globals/site.variables [hunk 7]
```diff
@@ -437,6 +437,7 @@
 /*-------------------
      Search Bar
 --------------------*/
+@datepickerDropdownMinWidth: 135px;
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/globals/site.variables]: This should be under the datepicker comment block or added to a more specific variables file.

File: src/styles/semantic/themes/livingstone/modules/dropdown.overrides [hunk 8]
```diff
@@ -183,17 +208,30 @@
     &.green ~ .ui.dropdown {
       padding-right: @dropdownWithIconPaddingRight;
     }
+
+    & ~ label {
+      width: @dropdownWithIconLabelWidth;
+
+      .dirty& {
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.overrides]: Is this intentional? Or are you missing a space?

File: src/styles/semantic/themes/livingstone/modules/dropdown.overrides [hunk 9]
```diff
@@ -183,17 +208,30 @@
     &.green ~ .ui.dropdown {
       padding-right: @dropdownWithIconPaddingRight;
     }
+
+    & ~ label {
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.overrides]: No need for the ampersand when writing these types of Less selectors

File: src/styles/semantic/themes/livingstone/modules/dropdown.overrides [hunk 10]
```diff
@@ -304,6 +350,21 @@
       right: @validIconRight;
       z-index: @validIconZIndex;
     }
+
+    & > i.icon {
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.overrides]: No need for the ampersand

File: src/styles/semantic/themes/livingstone/modules/dropdown.overrides [hunk 11]
```diff
@@ -304,6 +350,21 @@
...
+
+      &:not(.green) ~ label {
+        width: @validLabelWidthWithIcon;
+      }
+
+      ~ label {
+        width: @validLabelWidthWithoutIcon;
+
+        .dirty& {
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.overrides]: Missing a space?

File: src/styles/semantic/themes/livingstone/modules/dropdown.variables [hunk 12]
```diff
@@ -5,6 +5,7 @@
 /*-------------------
        Element
 --------------------*/
+@dropdownContainerMinWith: 90px;
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.variables]: typo `width`

File: src/styles/semantic/themes/livingstone/modules/dropdown.variables [hunk 13]
```diff
@@ -101,15 +102,22 @@
...
 
+@dropdownWithIconLabelWidth: ~'calc(100% - 75px)';
+
 @dropdownWithIconPaddingLeft: @40px;
 @dropdownWithIconPaddingRight: calc(@30px + @40px);
 
 /* With Image */
+
+@dropdownMinWidthWithImage: 118px;
+
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.variables]: No need for this new line

File: src/styles/semantic/themes/livingstone/modules/dropdown.variables [hunk 14]
```diff
@@ -101,15 +102,22 @@
 @iconPadding: @8px;
 @iconZIndex: @selectionZIndex + 1;
 
+@dropdownWithIconLabelWidth: ~'calc(100% - 75px)';
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.variables]: Be consistent, better to use double quotes

File: src/styles/semantic/themes/livingstone/modules/dropdown.variables [hunk 15]
```diff
@@ -101,15 +102,22 @@
...
 /* With Image */
+
+@dropdownMinWidthWithImage: 118px;
+
 @dropdownWithImagesCaretPaddingTop: @15px;
 
 /* With Label */
 @labelColor: @itemColor;
 
+@labelDefaultWidth: ~'calc(100% - 55px)';
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.variables]: We should stay consistent, better to stick with double quotes.

File: src/styles/semantic/themes/livingstone/modules/dropdown.variables [hunk 16]
```diff
@@ -165,17 +173,22 @@
...
+
 @validBorderColor: @green;
 
 @validIconOpacity: 1;
 @validIconPadding: @8px 0;
 @validIconRight: @40px;
 @validIconZIndex: 1;
 
+@validLabelWidthWithIcon: ~'calc(100% - 105px)';
+@validLabelWidthWithoutIcon: ~'calc(100% - 85px)';
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.variables]: For these two, they should also use double quotes.

File: src/styles/semantic/themes/livingstone/modules/dropdown.overrides [hunk 17]
```diff
@@ -11,6 +11,14 @@
   /*-------------------
        Content
 --------------------*/
+  > .text,
+  > .text .text,
+  > label {
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    overflow: hidden;
```
Comment (stephenkr) [inline src/styles/semantic/themes/livingstone/modules/dropdown.overrides]: I am seeing this used a lot throughout the PR, might be worth to add a detached rule set, such as:
```
@withTextOverflow: {
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
}
```
Forthe variables file.

And then in your stylesheet:
```
> label {
     @withTextOverflow;
}
```

You may need to check that we have the right Less version for this to work.

*http://lesscss.org/features/#detached-rulesets-feature

File: src/styles/semantic/themes/livingstone/modules/dropdown.overrides (hunk 8 above)
Comment (red-runner) [reply src/styles/semantic/themes/livingstone/modules/dropdown.overrides]: @stephenkr If you mean the & next to dirty, it is definitely intentional. Check the following chunk in https://lesstester.com/

```
.valid {
	
  ~ label {
  	background: red;
    
    .dirty& {
    	width: 100%;
    }
  }
}
```
**VS**
```
.valid {
	
  ~ label {
  	background: red;
    
    .dirty & {
    	width: 100%;
    }
  }
}
```

File: src/styles/semantic/themes/livingstone/modules/dropdown.overrides (hunk 11 above)
Comment (red-runner) [reply src/styles/semantic/themes/livingstone/modules/dropdown.overrides]: Same as here https://github.com/lodgify/lodgify-ui/pull/425#discussion_r231831516

File: src/styles/semantic/themes/livingstone/globals/site.variables (hunk 7 above)
Comment (red-runner) [reply src/styles/semantic/themes/livingstone/globals/site.variables]: @stephenkr I have checked on this but we do not have a file for `search.overrides` yet. Furthermore, `datepickers` are present both in `SearchBars` and `ContactForms` which makes it difficult to classify. Perhaps we could add a filed inside `site.overrides` called `datepicker`

**Pull request merged**